### DISCUSSION TITLE
How can I use this ?

### DISCUSSION THREAD

**Discussion by pullahs** [PlayFab/MultiplayerServerSecureWebsocket#2] [Q&A]
I'm not really good with network side of things. So, as far as I can assume, I'll just build this as a docker and run it on the server ? Then run the server docker build ( with a configured simple web transport) ? Or maybe should I combine this dockerfile with the one I created for the server build ?  Sorry for my terrible understanding (if any.)
(:

**Comment by AustinBrunkhorst** [comment]
> I'll just build this as a docker and run it on the server

[This Dockerfile](https://github.com/PlayFab/MultiplayerServerSecureWebsocket/blob/main/GameServer.ReverseProxy/Dockerfile) creates a HTTP server that handles requests at `/{buildId}/{sessionId}/{region}/{**forwardPath}`. Requests are proxied to a game server in the `Active` state that was requested with [Request Multiplayer Server](https://docs.microsoft.com/en-us/rest/api/playfab/multiplayer/multiplayer-server/request-multiplayer-server?view=playfab-rest).

https://github.com/PlayFab/MultiplayerServerSecureWebsocket/blob/27d9fa0b0db6a5cd2bcf7f2ba13fa0ba713fa7ee/GameServer.ReverseProxy/Startup.cs#L99

### Deployment
The docker application is intended to be deployed as a web service that is **separate** from your game server. Game servers deployed as HTTP hosts run into the same issue that the repository [MultiplayerServerSecureWebsocket](https://github.com/PlayFab/MultiplayerServerSecureWebsocket) mentions as a problem for games hosted on a https domain - you can't generate a SSL certificate for the fully qualified azure.com domain name of each server session.

[Azure App Service](https://docs.microsoft.com/en-us/azure/app-service/quickstart-dotnetcore?tabs=net60&pivots=development-environment-vs) is an "off the shelf" solution for the docker application. Linux and Windows should be supported since .NET Core is cross platform.

### Integrating with your web client
When your web client configures a web socket connection i.e. [ConnectEndpoint](https://docs-multiplayer.unity3d.com/docs/0.1.0/api/MLAPI.Transports.UNET.RelayTransport/#connectendpointint32-endpoint-int32-out-byte) with Unity or [WebSocket](https://developer.mozilla.org/en-US/docs/Web/API/WebSocket) in client side JavaScript, you'll use the fully qualified domain name from your deployed web service. 

For example, if you created an Azure App Service, the full qualified domain name would be something like `https://myreverseproxy.azurewebsites.net/{buildId}/{sessionId}/{region}/`
